cmd/orphan-finder/main.go

```diff
@@ -67,83 +98,142 @@ var (
-		return false, false
+
+	// The log line should contain a label indicating it is a cert or a precert
+	// orphan. We will determine which it is in checkDER based on the DER instead
+	// of the log line label.
+	if !strings.Contains(line, fmt.Sprintf("orphaning %s", certOrphan)) &&
+		!strings.Contains(line, fmt.Sprintf("orphaning %s", precertOrphan)) {
+		return false, false, unknownOrphan
+	}
+	// The log line should also contain certificate DER
+	if !strings.Contains(line, "cert=") {
+		return false, false, unknownOrphan
 	}
+	// Extract and decode the orphan DER
 	derStr := derOrphan.FindStringSubmatch(line)
 	if len(derStr) <= 1 {
 		logger.AuditErrf("Didn't match regex for cert: %s", line)
-		return true, false
+		return true, false, unknownOrphan
 	}
 	der, err := hex.DecodeString(derStr[1])
 	if err != nil {
 		logger.AuditErrf("Couldn't decode hex: %s, [%s]", err, line)
-		return true, false
+		return true, false, unknownOrphan
 	}
-	cert, err := checkDER(sa, der)
+	// Parse the DER, determine the orphan type, and ensure it doesn't already
+	// exist in the DB
+	cert, typ, err := checkDER(sa, der)
 	if err != nil {
 		logFunc := logger.Errf
 		if err == errAlreadyExists {
 			logFunc = logger.Infof
 		}
 		logFunc("%s, [%s]", err, line)
-		return true, false
+		return true, false, typ
 	}
 	// extract the regID
 	regStr := regOrphan.FindStringSubmatch(line)
 	if len(regStr) <= 1 {
 		logger.AuditErrf("regID variable is empty, [%s]", line)
-		return true, false
+		return true, false, typ
 	}
 	regID, err := strconv.Atoi(regStr[1])
 	if err != nil {
 		logger.AuditErrf("Couldn't parse regID: %s, [%s]", err, line)
-		return true, false
-	}
-	// OCSP-Updater will do the first response generation for this cert so pass an
-	// empty OCSP response. We use `cert.NotBefore` as the issued date to avoid
-	// the SA tagging this certificate with an issued date of the current time
-	// when we know it was an orphan issued in the past. Because certificates are
-	// backdated we need to add the backdate duration to find the true issued
-	// time.
-	issuedDate := cert.NotBefore.Add(backdateDuration)
+		return true, false, typ
+	}
+	// generate a fresh OCSP response
 	ocspResponse, err := ca.GenerateOCSP(ctx, core.OCSPSigningRequest{
 		CertDER: der,
 		Status:  string(core.OCSPStatusGood),
 	})
 	if err != nil {
 		logger.AuditErrf("Couldn't generate OCSP: %s, [%s]", err, line)
-		return true, false
+		return true, false, typ
+	}
+	// We use `cert.NotBefore` as the issued date to avoid the SA tagging this
+	// certificate with an issued date of the current time when we know it was an
+	// orphan issued in the past. Because certificates are backdated we need to
+	// add the backdate duration to find the true issued time.
+	issuedDate := cert.NotBefore.Add(backdateDuration)
+	switch typ {
+	case certOrphan:
+		_, err = sa.AddCertificate(ctx, der, int64(regID), ocspResponse, &issuedDate)
```

Comment:
Samesies here, if you use `strconv.ParseInt` rather than `strconv.Atoi` you get an int64.

Reply:
d989cef :ballot_box_with_check:  Thanks!